// constructor-classes/bundle.js
import Block from '../src/models/Block';
import Project from '../src/models/Project';
import Rollup from '../src/models/Rollup';
import Annotation from '../src/models/Annotation';
import Order from '../src/models/Order';
import Snapshot from '../src/models/Snapshot';

import fields from '../src/schemas/fields/index';

import BlockSchema from '../src/schemas/Block';
import ProjectSchema from '../src/schemas/Project';
import RollupSchema from '../src/schemas/Rollup';
import AnnotationSchema from '../src/schemas/Annotation';
import OrderSchema from '../src/schemas/Order';
import SnapshotSchema from '../src/schemas/Snapshot';

//there is an issue with isomorphic-fetch in web/node env, so force-polyfill fetch in this package
//so, we need something that explicitly modifies the global
//alternatively, if we know this package is going to be webpacked, then can add package.json "browser" field
//todo - include polyfill

export const models = {
  Block,
  Project,
  Rollup,
  Annotation,
  Order,
  Snapshot,
};

export { fields };

export const schemas = {
  Block: BlockSchema,
  Project: ProjectSchema,
  Rollup: RollupSchema,
  Annotation: AnnotationSchema,
  Order: OrderSchema,
  Snapshot: SnapshotSchema,
};
